docs/security.md

```diff
@@ -1,5 +1,54 @@
 # Security
```

Comment:
Given that this is doc on security, should we add a section on security policy of the project and link to https://github.com/argoproj/argo/blob/master/SECURITY.md?